WordPress/gutenberg
# [RNMobile] Media & Text - Media picker buttons functionality

Can be merged after https://github.com/WordPress/gutenberg/pull/17392

`gutenberg-mobile` PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/1378

## Description
This PR adds the missing functionality for the Media & Text block https://github.com/wordpress-mobile/gutenberg-mobile/issues/1313

On **Android** It will only support images until we can filter by more than one type at a time (images & videos) in the media picker.

Now the functionality it's the same as the `image` and `video` blocks.

## How has this been tested?

- Open the WordPress app and start a new post.
- Insert a new `Media & Text` block.
- Add an image from the photo library/camera.
- Expect the progress bar to show up while the file is being uploaded. While it's doing that you can cancel the upload.
- Once an image is uploaded you can do a long press to open the media options to change the file.
- If the connection fails, a retry option will be available.
- Test the image/video can be also selected from the WordPress media library.

## Screenshots

### iOS

<img src="https://user-images.githubusercontent.com/4885740/65469460-d353e800-de67-11e9-9cbf-2fb8d2133611.gif" width="200" />    <img src="https://user-images.githubusercontent.com/4885740/65469466-dbac2300-de67-11e9-8058-92cd8e59dcc0.gif" width="200" />

<img src="https://user-images.githubusercontent.com/4885740/65469468-de0e7d00-de67-11e9-9ca1-af512a217e9d.gif" width="200" />    <img src="https://user-images.githubusercontent.com/4885740/65469558-2af25380-de68-11e9-83bb-88c1d0fb703c.gif" width="200" />

<img src="https://user-images.githubusercontent.com/4885740/65469571-36de1580-de68-11e9-81cb-09159854dc0e.gif" width="200" />    <img src="https://user-images.githubusercontent.com/4885740/65469582-3f365080-de68-11e9-8cb3-8e4c2d9aef8b.gif" width="200" />

---

### Android

<img src="https://user-images.githubusercontent.com/4885740/65469654-773d9380-de68-11e9-8c9b-3e6e3d0c92f7.gif" width="200" />    <img src="https://user-images.githubusercontent.com/4885740/65469681-8e7c8100-de68-11e9-96a8-9ec43c0a9484.gif" width="200" />

<img src="https://user-images.githubusercontent.com/4885740/65469685-93d9cb80-de68-11e9-96c5-01dac37513dc.gif" width="200" />    <img src="https://user-images.githubusercontent.com/4885740/65469689-963c2580-de68-11e9-8aee-fec941f0a816.gif" width="200" />

---

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
